engine/src/main/java/org/terasology/logic/players/DebugControlSystem.java

```diff
@@ -74,24 +78,45 @@ public void onHideHUD(HideHUDButton event, EntityRef entity) {
         }
     }
 
+    /**
+     * Increases view distance upon receiving an increase view distance event.
+     * @param button The button or key pressed to increase view distance.
+     * @param entity The entity corresponding to event.
+     */
     @ReceiveEvent(components = ClientComponent.class)
     public void onIncreaseViewDistance(IncreaseViewDistanceButton button, EntityRef entity) {
         int viewDistance = config.getRendering().getViewDistance().getIndex();
         int maxViewDistance = ViewDistance.values().length - 1;
-
+        
+        //Ensuring that the view distance does not exceed its maximum value.
         if (viewDistance != maxViewDistance) {
-            config.getRendering().setViewDistance(ViewDistance.forIndex((viewDistance + 1)));
+            ViewDistance greaterViewDistance = ViewDistance.forIndex(viewDistance + 1);
+            config.getRendering().setViewDistance(greaterViewDistance);
+            fireChangeEvent("View distance increased to " + greaterViewDistance.toString() + ".", Arrays.asList(entity));	
+            //Presenting user with a warning if the view distance is set higher than recommended.
+            if (greaterViewDistance == ViewDistance.MEGA || greaterViewDistance == ViewDistance.EXTREME) {
+            	fireChangeEvent("Warning: Setting view distance to " + greaterViewDistance.toString() 
+            	                    + " may result in performance issues.", Arrays.asList(entity));
```

Comment:
Somewhat puzzled here why you wrap the entity in a List? :-)

It can only ever be one entity I'm thinking, and later on each element in the list is used for the event sending. Is there anything to it I'm missing, or maybe you started that way for some reason that later went away?